state/protocol/badger/mutator.go

```diff
@@ -85,28 +88,44 @@ func NewFullConsensusState(
 	blockTimer protocol.BlockTimer,
 	receiptValidator module.ReceiptValidator,
 	sealValidator module.SealValidator,
-) (*MutableState, error) {
+) (*ParticipantState, error) {
 	followerState, err := NewFollowerState(state, index, payloads, tracer, consumer, blockTimer)
 	if err != nil {
 		return nil, fmt.Errorf("initialization of Mutable Follower State failed: %w", err)
 	}
-	return &MutableState{
+	return &ParticipantState{
 		FollowerState:    followerState,
 		receiptValidator: receiptValidator,
 		sealValidator:    sealValidator,
 	}, nil
 }
 
-// Extend extends the protocol state of a CONSENSUS FOLLOWER. While it checks
+// ExtendCertified extends the protocol state of a CONSENSUS FOLLOWER. While it checks
 // the validity of the header; it does _not_ check the validity of the payload.
 // Instead, the consensus follower relies on the consensus participants to
-// validate the full payload. Therefore, a follower a QC (i.e. a child block) as
-// proof that a block is valid.
-func (m *FollowerState) Extend(ctx context.Context, candidate *flow.Block) error {
-
+// validate the full payload. Payload validity can be proved by a valid quorum certificate.
+// Certifying QC must match candidate block: candidate.View == certifyingQC.View && candidate.ID() == certifyingQC.BlockID
+// NOTE: this function expects that `certifyingQC` has been validated.
+// Expected errors during normal operations:
+//   - state.OutdatedExtensionError if the candidate block is outdated (e.g. orphaned)
+//   - state.InvalidExtensionError if the candidate block is invalid
```

Comment:
`InvalidExtensionError` is not an expected error now that we require a QC. `candidate` being invalid would constitute a critical bug or an exceeded Byzantine threshold. 
```suggestion
```

Reply:
https://github.com/onflow/flow-go/pull/3949/commits/a42afb0d3ad5d09d473b81730b0c13e41fd310b8